cmd/snapctl/plugin.go

```diff
@@ -116,6 +116,84 @@ func unloadPlugin(ctx *cli.Context) {
 	fmt.Printf("Type: %s\n", r.Type)
 }
 
+func swapPlugins(ctx *cli.Context) {
+	// plugin to load
+	pAsc := ctx.String("plugin-asc")
+	var paths []string
+	if len(ctx.Args()) < 1 || len(ctx.Args()) > 2 {
+		fmt.Println("Incorrect usage:")
+		cli.ShowCommandHelp(ctx, ctx.Command.Name)
+		os.Exit(1)
+	}
+	paths = append(paths, ctx.Args().First())
+	if pAsc != "" {
+		if !strings.Contains(pAsc, ".asc") {
+			fmt.Println("Must be a .asc file for the -a flag")
+			cli.ShowCommandHelp(ctx, ctx.Command.Name)
+			os.Exit(1)
+		}
+		paths = append(paths, pAsc)
+	}
+
+	// plugin to unload
+	var pDetails []string
+	var pType string
+	var pName string
+	var pVer int
+	var err error
+
+	if len(ctx.Args()) == 2 {
+		pDetails = filepath.SplitList(ctx.Args()[1])
+		if len(pDetails) == 3 {
+			pType = pDetails[0]
+			pName = pDetails[1]
+			pVer, err = strconv.Atoi(pDetails[2])
+			if err != nil {
+				fmt.Println("Can't convert version string to integer")
+				cli.ShowCommandHelp(ctx, ctx.Command.Name)
+				os.Exit(1)
+			}
```

Comment:
Would it be worthwhile to give an error message here so that inputs like 'collector:mock' don't get the error "Must provide plugin type"?